.github/workflows/docs.yml

```diff
@@ -0,0 +1,42 @@
+name: Docs
+
+on:
+  pull_request:
+    paths:
+      - 'docs/**'
+  push:
+    branches:
+      - master
+    paths:
+      - 'docs/**'
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    name: docs
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('docs/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+      - run: python -m pip install -r docs/requirements.txt
+      - name: Build docs
+        run: |
+          SPELLING_LOG=_build/spelling
+          cd docs
+          sphinx-build -b spelling -n -q -W --keep-going -d _build/doctrees -D language=en_US -j auto . $SPELLING_LOG
+          RETURN_CODE=$?
+          if [ $RETURN_CODE -ne 0 ]; then
+            echo "Spelling errors (please add real words to docs/spelling_wordlist):"
+            find $SPELLING_LOG -type f -name *.spelling | xargs cat
+          fi
+          exit $RETURN_CODE
```

Comment:
So this section doesn't run because GitHub Actions uses [`set -eo pipefail`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#exit-codes-and-error-action-preference) by default and the spelling errors aren't output. There are a few possible solutions:

1. Keep things as they are and use `shell: bash {0}` for this step to remove the fail-fast default.
2. Ditch this highlighted bit and set [`spelling_warning = True`](https://sphinxcontrib-spelling.readthedocs.io/en/latest/customize.html#output-options) in `conf.py`.
3. Ditch this highlighted bit and pass `-D spelling_warning=1` to `sphinx-build` above.

I prefer option 2 over option 3 as if we're treating spelling mistakes as errors in CI, then we should do the same locally. I don't like option 1 as, while it makes the current approach work, there is a built-in mechanism for doing what we want, so why do it manually?

I've boldly said this and then gone to check the [sphinxcontrib-spelling release notes](https://sphinxcontrib-spelling.readthedocs.io/en/latest/history.html#id1) and I see that it was actually added very recently in v7.1.0 released November 5, 2020. I don't see why we should just use this latest version, however, if it gives us exactly what we want.

Reply:
Thanks, much simpler.